rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/model/Transaction.java

```diff
@@ -720,61 +759,85 @@ private int lraStatusToHttpStatus() {
     public int begin(Long timeLimit) {
         int res = super.begin(); // no timeout because the default timeunit (SECONDS) is too course
 
-        startTime = LocalDateTime.now();
+        startTime = LocalDateTime.now(ZoneOffset.UTC);
 
         setTimeLimit(timeLimit);
 
+        deactivate();
+
         return res;
     }
 
     public int setTimeLimit(Long timeLimit) {
+        if (timeLimit <= 0L) {
+            return Response.Status.OK.getStatusCode();
+        }
+
         return scheduleCancelation(this::abortLRA, timeLimit);
     }
 
     private int scheduleCancelation(Runnable runnable, Long timeLimit) {
+        assert timeLimit > 0L;
+
         if (status() != ActionStatus.RUNNING) {
+            if (LRALogger.logger.isDebugEnabled()) {
+                LRALogger.logger.debugf("Ignoring timer because the action status is `%e'", status());
+            }
+
             return Response.Status.PRECONDITION_FAILED.getStatusCode();
         }
 
-        if (timeLimit > 0) {
-            if (finishTime != null) {
-                // check whether the new time limit is less than the current one
-                LocalDateTime ft = LocalDateTime.now().plusNanos(timeLimit * 1000000);
+        if (finishTime != null) {
+            // check whether the new time limit is less than the current one
+            LocalDateTime ft = LocalDateTime.now().plusNanos(timeLimit * 1000000);
```

Comment:
sorry I missed this previously. This is missing `ZoneOffset.UTC`.

Reply:
+1